geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/vector/SimpleFeatureVector.scala

```diff
@@ -245,7 +252,18 @@ object SimpleFeatureVector {
       val isLong = dateVector.exists(_.isInstanceOf[BigIntVector])
       if (isLong) { Encoding.Max } else { Encoding.Min }
     }
-    val encoding = SimpleFeatureEncoding(fidEncoding, geomPrecision, datePrecision)
+    val axisOrderPrecision = {
+      if (java.lang.Boolean.getBoolean("org.geotools.referencing.forceXY") ||
+        Hints.getSystemDefault(Hints.FORCE_LONGITUDE_FIRST_AXIS_ORDER) == java.lang.Boolean.TRUE) {
```

Comment:
I think this will always be true, as we set it globally here: https://github.com/locationtech/geomesa/blob/main/geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/geotools/MetadataBackedDataStore.scala#L49

the exception would be PostGIS I guess.

How did you decide that this would be the way to check?

Reply:
Bit of a contradiction then: the current Arrow logic is hard coded to use Lat/Lon (Y/X) order but the `MetadataBackendDataStore` hint is stating that it should always be the reverse (Lon/Lat, X/Y) order. Not seeing the hint being referenced in the `geomesa` repo, can / should it be removed at this point?

> How did you decide that this would be the way to check?

Think I read through this note:
https://docs.geotools.org/stable/userguide/library/referencing/order.html

and given that I didn't see any way to extract the encoding from a `StructVector` or it's derived `SimpleFeatureType` ended up falling back to the environment variables and hints.

Is there a better way to derive the flag here in `SimpleFeatureVector.getFeatureType(vector: StructVector)`?

Reply:
yes, we weren't entirely consistent in the axis order :sweat_smile: . we have a TODO to remove the longitude first hint, but I'm unsure of the repercussions. i think it would probably be easiest to default to lat/lon in arrow (which is the current behavior), and only flip to lon/lat if the hint is explicitly provided. there could still be cases where someone expects a certain axis order based on the CRS, and it won't automatically work, but I'd say that's something we can save for later if it ever comes up.

Reply:
Yup so kept the existing behavior as the default and the encoders just referencing a derived boolean.

Not thrilled that there's no way to derive the axis order from the `StructVector` but from my manual testing this doesn't seems to be an issue at this point.